sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -86,22 +84,17 @@ def _wrap_files_in_tarball(self, tarball_path, files={}):
       for key, value in files.items():
         tarball.add(value, arcname=key)
 
-  def prepare_docker_tarball_with_py(self, arc_python_filename, python_filepath, base_image):
+  def prepare_docker_tarball_with_py(self, arc_python_filename, python_filepath, base_image, local_tarball_path):
```

Comment:
How about `prepare_docker_tarball_with_py(self, python_filepath, local_tarball_path, base_image='python:3.5', arc_python_filename='program.py')`

BTW, can `python_filepath` be absolute? Will this work?

Reply:
>Good question, the python_filepath can be absolute or relative.

That's good. The problem is that your code will fail in one of those cases.

Reply:
>the error messages might be misleading once the container launched but the program.py is not found.
How come this file is not in the container? You're putting it there yourself.

Reply:
You are right. I'm good with default value for arc_python_filename. However, I do not think we should add a default value for base_image. This parameter should be mandatory from the DockerfileHelper if not for out layer APIs.